apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/AccountControlController.java

```diff
@@ -0,0 +1,462 @@
+            @FormParam("recipientPublicKey") String recipientPublicKey,
+        @Parameter @Schema(description = "ec Block Id", implementation = Long.class)
+            @FormParam("ecBlockId") Long ecBlockId,
+        @Parameter @Schema(description = "ec Block Height", implementation = Integer.class)
+            @FormParam("ecBlockHeight") Integer ecBlockHeight,
+
+        @Parameter @Schema(description = "Passphrase to vault account, should be specified if sender account is vault")
+            @FormParam("passphrase") String passphrase,
+        @Parameter @Schema(description = "Two-factor auth code, if 2fa enabled")
+            @FormParam("code2FA") @DefaultValue("0") Integer code2FA,
+        @Context HttpServletRequest servletRequest
+    ) {
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        log.trace("Started leaseBalance, recipientIdParameter = {}, senderIdParameter={}, period={}",
+            recipientIdParameter, senderIdParameter, period);
+
+        long recipientAccountId = recipientIdParameter.get();
+        Account recipientAccount = accountService.getAccount(recipientAccountId);
+        log.trace("recipientAccountId = {}, recipientAccount={}", recipientAccountId, recipientAccount);
+        if (recipientAccount == null || accountService.getPublicKeyByteArray(recipientAccountId) == null) {
+            return response.error(ApiErrors.CUSTOM_ERROR_MESSAGE,
+                "recipient account not found OR it does not have public key").build();
+        }
+        long senderAccountId = senderIdParameter.get();
+        Account senderAccount = accountService.getAccount(senderAccountId);
+        log.trace("senderAccountId = {}, senderAccount={}", senderAccountId, senderAccount);
+
+        Attachment attachment = new AccountControlEffectiveBalanceLeasing(period);
+
+        CreateTransactionRequest txRequest;
+        try {
+            txRequest = HttpRequestToCreateTransactionRequestConverter.convert(
+                servletRequest, senderAccount, recipientAccount, recipientAccountId, 0, feeATM, attachment, broadcast);
+            log.trace("txRequest = {}", txRequest);
+        } catch(ParameterException e) {
+            log.warn("Tx conversion param exception", e);
+            return response.error(ApiErrors.CUSTOM_ERROR_MESSAGE, e.getErrorResponse()).build();
+        } catch(Exception e) {
+            log.warn("Tx global exception", e);
+            return response.error(ApiErrors.CUSTOM_ERROR_MESSAGE, "global exception: " + e.getMessage()).build();
+        }
+
+        Transaction transaction = txCreator.createTransactionThrowingException(txRequest);
+        log.trace("leaseBalance transaction = {}", transaction);
+        UnconfirmedTransactionDTO txDto = unconfirmedTransactionConverter.convert(transaction);
+        log.trace("leaseBalance txDto = {}", txDto);
+        LeaseBalanceResponse leaseBalanceResponse = new LeaseBalanceResponse();
```

Comment:
Maybe move such operations to converter?

Reply:
I will see if that json structure is used fir many methods... if that's true then it's worth to create converter.